File: articles/systems-foresight-and-structural-change/julia/systems_foresight_score.jl
# Julia standard-library example for structural pressure scoring.

systems = Dict(
    "Climate Adaptation" => [0.86, 0.46, 0.52, 0.88, 0.90],
    "Institutional Trust" => [0.76, 0.38, 0.36, 0.78, 0.82],
    "Food-Water-Ecology" => [0.84, 0.48, 0.54, 0.90, 0.84]
)

function structural_pressure(v)
    0.26*v[1] + 0.22*(1-v[2]) + 0.18*(1-v[3]) + 0.18*v[4] + 0.16*v[5]
end

println("system,structural_pressure")
for (name, values) in sort(collect(systems))
    println("$(name),$(round(structural_pressure(values), digits=4))")
end

File: articles/systems-foresight-and-structural-change/cpp/systems_foresight_score.cpp
#include <iostream>
#include <vector>

double structural_pressure(const std::vector<double>& v) {
    return 0.26 * v[0] + 0.22 * (1.0 - v[1]) + 0.18 * (1.0 - v[2]) + 0.18 * v[3] + 0.16 * v[4];
}

int main() {
    std::vector<double> climate_adaptation = {0.86, 0.46, 0.52, 0.88, 0.90};
    std::cout << "Climate adaptation structural pressure="
              << structural_pressure(climate_adaptation) << "\n";
    return 0;
}
